_posts/2022-02-09-reviewing-pull-requests.adoc

```diff
@@ -0,0 +1,182 @@
+
+image::/assets/img/reviewing-pull-requests/no-problems.png[No problems reported by the Java plugin]
+Figure 7: No problems reported by the Java plugin in the Problems view.
+
+=== Running unit tests and building
+The devfile also defines commands for testing, building, and launching the application.
+Let's run the unit tests by opening the Workspace view from the right-hand side and clicking `(User Runtimes -> tools -> runtests)`.
+
+This runs the test command (`./mvnw test`) within the `tools` container as specified in the devfile. The test output can be viewed in the output panel.
+
+image::/assets/img/reviewing-pull-requests/run-tests.png[Running the unit tests]
+Figure 8: Unit testing by clicking `runtests` from the Workspace view on the right.
+
+As we can see in the output from Figure 8, all of the tests pass!
+
+You can also run other commands such as `(User Runtimes -> tools -> package)` to build the application.
+
+image::/assets/img/reviewing-pull-requests/build.png[Successfully building the application]
+Figure 9: Successfully building the application.
+
+NOTE: Please note, the `packagenative` command is used to build a native image with GraalVM. This command would fail for our example project on link:https://www.eclipse.org/che/docs/che-7/hosted-che/hosted-che/[Eclipse Che Hosted by Red Hat] due to the 7GB memory usage limit.
+
+=== Running the application
+Let's run the Quarkus application in link:https://quarkus.io/guides/getting-started#development-mode[development mode] by running the `(User Runtimes -> tools -> startdev)` command to try accessing the endpoint ourself.
+
+image::/assets/img/reviewing-pull-requests/start-dev.png[Starting the application in development mode]
+Figure 10: Starting the Quarkus project in development mode by clicking `startdev` from the Workspace view on the right.
+
+Next, let's access the new `/food/restaurant/{restaurantName}` endpoint. Here, we access `/food/restaurant/Local Deli` to get all `Food` resources from the restaurant named `Local Deli`. 
+
+image::/assets/img/reviewing-pull-requests/access-endpoint.png[Accessing the new endpoint]
+Figure 11: Response from `/food/restaurant/Local Deli`.
+
+There are two `Food` resources from the `Local Deli` restaurant from the response, which is what we would expect given that these are the only `Food` resources from the specified restaurant in our link:https://github.com/che-incubator/quarkus-api-example/blob/main/src/main/resources/import.sql[`import.sql`] file.
+
+=== Providing feedback and merging the PR
+So far, we have successfully run tests, ran the build, as well as ran the application in development mode to verify that the PR is working correctly.
+Next, lets merge the PR from the IDE.
+
+Going back to the GitHub PR view (see Figure 5), we can provide any additional comments, and approve the PR. Let's merge this PR to `main`.
```

Comment:
btw, I do not think it is possible to approve the PRs due to the restrictions in the `che-incubator`NOTE org

![image](https://user-images.githubusercontent.com/1461122/153602462-1036e17f-dca3-40da-bebc-abd233ff7ea3.png)

@benoitf @l0rd folks, wdyt can we disable the OAuth App access restrictions for the che-incubator org?

Reply:
![image](https://user-images.githubusercontent.com/436777/153896638-782695c1-fecc-4502-88d8-027a6d4c8929.png)

So I approved the request for devsandbox oAuth, let me try if it works for you now

Reply:
@benoitf seems to have the very same issue, I think we need to approve both clusters [host] and [stage], not just [member]

Reply:
thanks for enabling the access for all clusters, can approve PRs from IDE now \o/

![image](https://user-images.githubusercontent.com/1461122/154050800-4c399618-8439-4a8f-b257-5bb0ec74609a.png)

Reply:
closing the issue as it's been resolved